pkg/ui/workspaces/cluster-ui/src/highlightedText/highlightedText.tsx

```diff
@@ -83,6 +86,12 @@ export function getHighlightedText(
   const search = highlight
     .split(" ")
     .map(val => {
+      // When it's only a character we want to make sure to use the literal value (by adding []),
+      // this is added to handle cases where the highlighted term is a character used normally
+      // in regex, such as '*', '+', '?'
+      if (val.length == 1) {
```

Comment:
This is likely to produce a strange result if the character is `\` or `^`.
Are you sure about this?

Reply:
A regular expression with the syntax `[\]` is incorrect: it's malformed. The proper syntax should be `[\\]` if you want to match `\`.

A regular expression with the syntax `[^]` matches "everything except the characters after `^`", in this case there is o character so `[^]` is equivalent to `.` (i.e. match anything). The proper syntax to match `^` is ... just `^` or `[\^]`.